app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreenViewModel.kt

```diff
@@ -63,50 +80,82 @@ class OtherUserProfileScreenViewModel @Inject constructor(
+                    println("cyka 3")
+                    appLogger.d("Couldn't not find the user with provided id: $userId")
+                    snackBarState = SnackBarState.LoadUserInformationError()
                 }
-                is GetUserInfoResult.Success ->
+                conversationResult is CreateConversationResult.Failure -> {
+                    println("cyka 2")
+                    appLogger.d("Couldn't not getOrCreateOneToOneConversation for user id: $userId")
+                    snackBarState = SnackBarState.LoadUserInformationError()
+                }
+                conversationResult is CreateConversationResult.Success && userInfoResult is GetUserInfoResult.Success -> {
+                    println("cyka 1 $conversationId")
                     conversationId
                         .let { if (it != null) observeConversationRoleForUser(it, userId) else flowOf(it) }
-                        .collect { loadViewState(result.otherUser, result.team, it) }
+                        .combine(observeSelfUser(), ::Pair)
+                        .collect { (conversationRoleData, selfUser) ->
+                            loadViewState(userInfoResult, conversationResult.conversation, conversationRoleData, selfUser.teamId)
+                        }
+                }
             }
         }
     }
 
-    private fun loadViewState(otherUser: OtherUser, team: Team?, conversationRoleData: ConversationRoleData?) {
+    private fun loadViewState(
+        getInfoResult: GetUserInfoResult.Success,
+        directConversation: Conversation,
+        conversationRoleData: ConversationRoleData?,
+        selfTeamId: TeamId?
+    ) {
+        val userAvatarAsset = getInfoResult.otherUser.completePicture
+            ?.let { pic -> ImageAsset.UserAvatarAsset(wireSessionImageLoader, pic) }
+
         state = state.copy(
             isDataLoading = false,
-            userAvatarAsset = otherUser.completePicture?.let { pic -> ImageAsset.UserAvatarAsset(wireSessionImageLoader, pic) },
-            fullName = otherUser.name ?: String.EMPTY,
-            userName = mapUserLabel(otherUser),
-            teamName = team?.name ?: String.EMPTY,
-            email = otherUser.email ?: String.EMPTY,
-            phone = otherUser.phone ?: String.EMPTY,
-            connectionStatus = otherUser.connectionStatus,
-            membership = userTypeMapper.toMembership(otherUser.userType),
+            userAvatarAsset = userAvatarAsset,
+            fullName = getInfoResult.otherUser.name.orEmpty(),
+            userName = mapUserLabel(getInfoResult.otherUser),
+            teamName = getInfoResult.team?.name.orEmpty(),
+            email = getInfoResult.otherUser.email.orEmpty(),
+            phone = getInfoResult.otherUser.phone.orEmpty(),
+            connectionStatus = getInfoResult.otherUser.connectionStatus,
+            membership = userTypeMapper.toMembership(getInfoResult.otherUser.userType),
```

Comment:
Not important, but maybe we could add `with(getInfoResult)` on top to avoid recurrent field accessor calls